tests/test_snapshot/test_file.py

```diff
@@ -220,3 +222,20 @@ def test_write(
         assert snapshot_file.exists() == file_written
         if file_written:
             assert snapshot_file_object._snapshot_statuses == expected_statuses
+
+    @staticmethod
+    def test_encoded_metadata(config: Config, snapshot_file: Path):
+        """ Test that SnapshotMetadata gets encoded to eliminate serialization errors. """
+        # Arrange
+        args = dict(unhandled_type=SimpleNamespace(complex=3 + 4j))
+        caller_info = CallerInfo(snapshot_file, "test_function", args)
+        metadata = SnapshotMetadata(caller_info, update_on_next_run=False)
+        snapshot_file_object = _SnapshotFile(config, metadata)
+
+        # Act
+        snapshot_file_object._changed_flag = True
+        snapshot_file_object.write()
```

Comment:
The created snapshot is cleanup properly after running tests correct?

Reply:
Yes, the `snapshot_file` uses the `pytest` temporary directory